pkg/controllers/jobexecution_controller.go

```diff
@@ -116,6 +118,16 @@ func (r *JobExecutionReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{RequeueAfter: time.Second * 15}, nil
 }
 
+// Returns true if there is at least one condition from the job that has the failed status.
+func hasFailedCondition(job *batchv1.Job) bool {
+	for _, c := range job.Status.Conditions {
+		if c.Type == batchv1.JobFailed {
```

Comment:
Should we also consider the Failure reason? e.g. `reason: BackoffLimitExceeded`

We have `backoffLimit: 0` as default, but JobExecution would possibly prematurely fail on the first attempty if `backoffLimit > 0`.

Reply:
I tested it with zero but also with four, and it works. Not sure what we should be adding

Reply:
I tested it now too and works right 👍🏽 I just thought there would be other `type: Failed` causes that could lead to a premature termination of the JobExecution.

All good. Resolving.